docs/guides/func_yaml.md

```diff
@@ -0,0 +1,88 @@
+# Project Configuration with `func.yaml`
+
+The `func.yaml` file contains configuration information for your function
+project. Generally, these values are used when you execute a `func` CLI
+command. For example, when `func build` is run, the CLI uses the value for
+the `builder` field. In many cases, these values may be overridden by
+command line flags or environment variables. For more information about
+overriding these values, consult the [Commands](command.md) document.
+
+Many of the fields are generated for you when you create, build and deploy
+your function. However there are a few that you may use to tweak things
+such as the function name, and the image name.
+
+## Fields
+
+The following fields are used in `func.yaml`.
+
+### `builder`
+
+Specifies the buildpack builder image to use when building the function.
+In most cases, this value should not be changed.
+
+### `builderMap`
+
+Some function runtimes may be built in multiple ways. For example, a Quarkus
+function may be built for the JVM, or as a native binary. The `builderMap`
+field will contain all of the available builders for a given runtime. Although
+it's typically unwise to modify the `builder` field, using values from
+`builderMap` is OK.
+
+### `envVars`
+
+The `envVars` field allows you to set environment variables that will be
+available to your function at runtime. For example, to set a `MODE` environment
+variable to `debug` when the function is deployed, your `func.yaml` file
+may look like this.
+
+```yaml
+envVars: {
+  MODE: debug
+}
```

Comment:
While I much prefer the use of delimiters over indentation, it seems to be more common to use the latter in YAML files, despite this being perfectly valid.  So, should this instead be:
```
envVars:
  MODE: debug
```